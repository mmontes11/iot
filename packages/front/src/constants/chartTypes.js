export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
export const REALTIME = "REALTIME";
