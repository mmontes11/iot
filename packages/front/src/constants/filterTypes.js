export const THING_FILTER_TYPE = "thing";
export const DATE_FILTER_TYPE = "date";
export const FILTER_TYPES = [THING_FILTER_TYPE, DATE_FILTER_TYPE];
