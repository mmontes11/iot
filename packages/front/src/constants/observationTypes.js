export const EVENT_TYPE = "event";
export const MEASUREMENT_TYPE = "measurement";
export const OBSERVATION_TYPES = [EVENT_TYPE, MEASUREMENT_TYPE];
