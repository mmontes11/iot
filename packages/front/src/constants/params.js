export const TYPE = "TYPE";
export const OBSERVATION = "OBSERVATION";
export const GROUPBY = "GROUPBY";
export const THING = "THING";
