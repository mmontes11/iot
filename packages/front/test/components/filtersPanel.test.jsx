import React from "react";
import { shallow } from "enzyme";
import FiltersPanel from "components/filtersPanel";
import { FILTER_TYPES } from "constants/filterTypes";

describe("components/filterPanel", () => {
  it("renders with no filters", () => {
    const wrapper = shallow(
      <FiltersPanel
        type={{
          items: ["foo", "bar"],
          isActive: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
        }}
        thingFilter={{
          label: "Select filter:",
          items: ["foo", "bar"],
          isActive: false,
          isLoading: false,
          isDisabled: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
          onDelete: () => undefined,
        }}
        dateFilter={{
          selector: {
            isCustomSelected: false,
            onChange: () => undefined,
          },
          timePeriod: {
            label: "Select time period:",
            isLoading: false,
            isActive: true,
            isDisabled: false,
            items: ["foo", "bar"],
            selectedItem: null,
            onButtonClick: () => undefined,
            onItemClick: () => undefined,
          },
          custom: {
            startDate: {
              selected: null,
              onChange: () => undefined,
            },
            endDate: {
              selected: null,
              onChange: () => undefined,
            },
          },
          onDelete: () => undefined,
        }}
        selectedFilters={["foo"]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders with filters", () => {
    const wrapper = shallow(
      <FiltersPanel
        type={{
          items: ["foo", "bar"],
          isActive: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
        }}
        thingFilter={{
          label: "Select filter:",
          items: ["foo", "bar"],
          isActive: false,
          isLoading: false,
          isDisabled: false,
          onButtonClick: () => undefined,
          onItemClick: () => undefined,
          onDelete: () => undefined,
        }}
        dateFilter={{
          selector: {
            isCustomSelected: false,
            onChange: () => undefined,
          },
          timePeriod: {
            label: "Select time period:",
            isLoading: false,
            isActive: true,
            isDisabled: false,
            items: ["foo", "bar"],
            selectedItem: null,
            onButtonClick: () => undefined,
            onItemClick: () => undefined,
          },
          custom: {
            startDate: {
              selected: null,
              onChange: () => undefined,
            },
            endDate: {
              selected: null,
              onChange: () => undefined,
            },
          },
          onDelete: () => undefined,
        }}
        selectedFilters={FILTER_TYPES}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
