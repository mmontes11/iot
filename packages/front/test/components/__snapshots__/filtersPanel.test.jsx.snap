// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/filterPanel renders with filters 1`] = `
<div
  className="box"
>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <Dropdown
        buttonStyle="is-primary"
        iconStyle="fa-plus"
        isActive={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Filters"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      />
    </div>
  </div>
  <ThingFilter
    key="thing"
    thingFilter={
      Object {
        "isActive": false,
        "isDisabled": false,
        "isLoading": false,
        "items": Array [
          "foo",
          "bar",
        ],
        "label": "Select filter:",
        "onButtonClick": [Function],
        "onDelete": [Function],
        "onItemClick": [Function],
      }
    }
  />
  <InjectIntl(DateFilter)
    dateFilter={
      Object {
        "custom": Object {
          "endDate": Object {
            "onChange": [Function],
            "selected": null,
          },
          "startDate": Object {
            "onChange": [Function],
            "selected": null,
          },
        },
        "onDelete": [Function],
        "selector": Object {
          "isCustomSelected": false,
          "onChange": [Function],
        },
        "timePeriod": Object {
          "isActive": true,
          "isDisabled": false,
          "isLoading": false,
          "items": Array [
            "foo",
            "bar",
          ],
          "label": "Select time period:",
          "onButtonClick": [Function],
          "onItemClick": [Function],
          "selectedItem": null,
        },
      }
    }
    key="date"
  />
</div>
`;

exports[`components/filterPanel renders with no filters 1`] = `
<div
  className="box"
>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <Dropdown
        buttonStyle="is-primary"
        iconStyle="fa-plus"
        isActive={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Filters"
        onButtonClick={[Function]}
        onItemClick={[Function]}
      />
    </div>
  </div>
</div>
`;
